applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -248,16 +357,42 @@ static void stream_configured_cb(struct bt_audio_stream *stream,
 		return;
 	}
 
+	new_pd_us = find_headset_pres_delay(CONFIG_BT_AUDIO_PRES_DLY_SRCH_MODE, channel_index);
+	if (new_pd_us == 0) {
+		LOG_ERR("Cannot get a valid presentation delay");
+	}
+
+	LOG_DBG("New presentation delay: %d us", new_pd_us);
+	lc3_preset_sink.qos.pd = new_pd_us;
+
 #if CONFIG_STREAM_BIDIRECTIONAL
 	if (ep_state_check(headsets[channel_index].sink_stream.ep,
 			   BT_AUDIO_EP_STATE_CODEC_CONFIGURED) &&
 	    ep_state_check(headsets[channel_index].source_stream.ep,
 			   BT_AUDIO_EP_STATE_CODEC_CONFIGURED))
-#endif /* CONFIG_STREAM_BIDIRECTIONAL */
+#endif
 	{
+		for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+			if (i != channel_index) {
+				if (headsets[i].headset_conn != NULL &&
+				    headsets[i].sink_stream.ep != NULL) {
+					ret = update_sink_stream_qos(&headsets[i], new_pd_us);
+					if (ret) {
+						LOG_ERR("Failed to set QoS for stream: %d", ret);
+					}
+				}
+			}
+		}
+
+		LOG_DBG("Set %s headset, connection %p, stream %p", headsets[channel_index].ch_name,
+			&headsets[channel_index].headset_conn,
+			&headsets[channel_index].sink_stream);
+
+		headsets[channel_index].sink_stream.qos->pd = new_pd_us;
```

Comment:
Hmm, I am not sure we should set the QOS parameters from the application like this

Reply:
Do you have a suggestion we could discuss?

Reply:
It seems like this is the correct way to do it after all